pkg/controller/core/localqueue_controller.go

```diff
@@ -174,11 +207,50 @@ func (h *qWorkloadHandler) Generic(e event.GenericEvent, q workqueue.RateLimitin
 	q.AddAfter(req, constants.UpdatesBatchPeriod)
 }
 
+// qCQHandler signals the controller to reconcile the Queue associated
+// to the workload in the event.
+// Since the events come from a channel Source, only the Generic handler will
+// receive events.
+type qCQHandler struct {
+	qManager *queue.Manager
+}
+
+func (h *qCQHandler) Create(event.CreateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) Update(event.UpdateEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) Delete(event.DeleteEvent, workqueue.RateLimitingInterface) {
+}
+
+func (h *qCQHandler) Generic(e event.GenericEvent, q workqueue.RateLimitingInterface) {
+	cq, ok := e.Object.(*kueue.ClusterQueue)
+	if !ok {
+		return
+	}
+	qKeys := h.qManager.LocalQueuesReferenceClusterQueue(cq.Name)
```

Comment:
better list from the client, which has an index

Reply:
Done.